projects/hg-resolvers/src/lib/resolver.ts

```diff
@@ -200,7 +200,7 @@ export class Resolver<T, D = any> {
         // if we have isFunctionObservableTarget === false as we are in the state of resolving
         target.cancelAction();
       }
-      if (this.previousState !== null) { this.state = this.previousState; }
+      if (this.previousState !== null && this.state !== ResolverState.SETTLED) { this.state = this.previousState; }
```

Comment:
From what I see this is correct but shouldn't we also include the ERRORED state. 
`this.previousState !== null && [ResolverState.SETTLED, ResolverState.ERRORED].includes(this.state)`

Reply:
Make sense, i will extend it.